reward-monitor-lambda-app/src/utils.ts

```diff
@@ -0,0 +1,42 @@
+import fetch from 'node-fetch';
+import pRetry from 'p-retry';
+
+/**
+ * Send a PagerDuty event via PagerDuty Event v2 API
+ * @param message message to send to PagerDuty
+ * @param severity severity of the event. Can be one of: 'info', 'warning', 'error', 'critical'
+ */
+export async function sendPagerDutyEvent(message: string, severity: string) {
+    // Create a PagerDuty Event API payload
+    const payload = {
+        routing_key: process.env.PAGERDUTY_ROUTING_KEY,
+        event_action: 'trigger',
+        payload: {
+            summary: message,
+            severity: severity,
+            source: 'reward-monitor-lambda-app',
+        },
+    };
+
+    // Send the payload to PagerDuty Event API
+    // Use pRetry to retry the request 3 times
+    const response = await pRetry(
+        () =>
+            fetch('https://events.pagerduty.com/v2/enqueue', {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify(payload),
+            }),
+        {
+            retries: 3,
+        },
+    ).catch((err) => {
+        console.error(`Failed to send PagerDuty event. Error: ${err}`);
+        throw err;
```

Comment:
what happens to thrown errors in SAM?

Reply:
If a thrown error is uncaught, Lambda functions will stop executing and return data like this
https://docs.aws.amazon.com/lambda/latest/dg/nodejs-exceptions.html

The activities (and logs) are stored so we can monitor them ourselves or using aws dashboard
https://docs.aws.amazon.com/lambda/latest/dg/lambda-monitoring.html
https://docs.aws.amazon.com/lambda/latest/dg/monitoring-insights.html

But since we want to use the lambda to serve as an API, we need to catch them.
If we dont catch anything, API gateway will just return arbitrary 500 response "Internal server error" like below.
```
HTTP/1.1 500 Internal Server Error
Content-Type: application/json
{
"Message": "Internal server error"
}
```

So we should catch errors at root level and make sure the lambda function always returns an object with status codes (2xx, 4xx, 5xx) w/ meaningful messages or stack traces.